index.js

```diff
@@ -1,10 +1,99 @@
 // import your node modules
 
 const db = require('./data/db.js');
-const server = require('express')();
+const express = require('express');
 
 // add your server code starting here
 
+const server = express();
+
+server.use(express.json());
+
+server.post('/api/posts', (req, res) => {
+
+  const post = req.body;
+
+  if (!post.title || !post.contents) {
+
+    res.status(400).json({ errorMessage: "Please provide title and contents for the post." });
+
+  }
+
+  else {
+
+    db.insert(post)
+      .then(data => res.status(201).json(data))
+      .catch(err => res.status(500).json({ error: "There was an error while saving the post to the database" }));
+
+  }
+
+});
+
+server.delete('/api/posts/:id', async (req, res) => {
+
+  try {
+
+    const post = await db.findById(req.params.id);
+
+    if (post.length) {
+
+      await db.remove(req.params.id);
+
+      res.status(200).json(post).end();
```

Comment:
Thanks to our revelation in standup, we don't need `end` here! `json` (and `send`) will do that for us.